### PR TITLE
Verified domain admin annotations

### PR DESCRIPTION
**Pull request by jacekwegr**
Check for incorrect or missing admin annotations in GraphQL schemas

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/132326) / small_tests / b8abab8a3f2d28f62b318105719beef496ecb802
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3731/132326/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3731/132326/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-08-11_14.58.45/lib.mongooseim.logs/run.2022-08-11_14.58.45/suite.log.html)

File: priv/graphql/schemas/admin/muc.gql [hunk 1]
```diff
@@ -7,13 +7,13 @@ type MUCAdminMutation @protected{
     @protected(type: DOMAIN, args: ["mucDomain"])
   "Invite a user to a MUC room"
   inviteUser(room: JID!, sender: JID!, recipient: JID!, reason: String): String
-    @protected(type: DOMAIN, args: ["sender"])
+    @protected(type: DOMAIN, args: ["room"])
```
Comment (JanuszJakubiec) [inline priv/graphql/schemas/admin/muc.gql:10]: I think that the domain should be taken both from room and sender.

File: priv/graphql/schemas/admin/muc.gql [hunk 2]
```diff
@@ -7,13 +7,13 @@ type MUCAdminMutation @protected{
...
   "Kick a user from a MUC room"
   kickUser(room: JID!, nick: String!, reason: String): String
     @protected(type: DOMAIN, args: ["room"])
   "Send a message to a MUC room"
   sendMessageToRoom(room: JID!, from: FullJID!, body: String!): String
-    @protected(type: DOMAIN, args: ["from"])
+    @protected(type: DOMAIN, args: ["room"])
```
Comment (JanuszJakubiec) [inline priv/graphql/schemas/admin/muc.gql:16]: I think that the domain should be taken both from the room and "from".

File: priv/graphql/schemas/admin/muc.gql [hunk 3]
```diff
@@ -31,10 +31,10 @@ type MUCAdminMutation @protected{
     @protected(type: DOMAIN, args: ["room"])
   "Make a user enter the room with a given nick"
   enterRoom(room: JID!, user: FullJID!, nick: String!, password: String): String
-    @protected(type: DOMAIN, args: ["user"])
+    @protected(type: DOMAIN, args: ["room"])
```
Comment (JanuszJakubiec) [inline priv/graphql/schemas/admin/muc.gql:34]: I think that the domain should be taken both from the room and user.

File: priv/graphql/schemas/admin/muc.gql [hunk 4]
```diff
@@ -31,10 +31,10 @@ type MUCAdminMutation @protected{
...
   "Make a user with the given nick exit the room"
   exitRoom(room: JID!, user: FullJID!, nick: String!): String
-    @protected(type: DOMAIN, args: ["user"])
+    @protected(type: DOMAIN, args: ["room"])
```
Comment (JanuszJakubiec) [inline priv/graphql/schemas/admin/muc.gql:37]: I think that the domain should be taken both from the room and user.

File: priv/graphql/schemas/admin/muc_light.gql [hunk 5]
```diff
@@ -7,10 +7,10 @@ type MUCLightAdminMutation @protected{
     @protected(type: DOMAIN, args: ["mucDomain"])
   "Change configuration of a MUC Light room"
   changeRoomConfiguration(room: JID!, owner: JID!, name: String!, subject: String!): Room
-    @protected(type: DOMAIN, args: ["room", "owner"])
```
Comment (JanuszJakubiec) [inline priv/graphql/schemas/admin/muc_light.gql:10]: I think it's better to have more than fewer variables from which we take the domain. That way we are safe against some edge cases when the room has an owner from a different domain

**Review by JanuszJakubiec — COMMENTED** [review]
Looks good, I've added some minor comments.

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/132627) / small_tests / a59756f6c1aea02fdcebc5b44888592f0d58c762
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3731/132627/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3731/132627/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-08-12_13.54.15/lib.mongooseim.logs/run.2022-08-12_13.54.16/suite.log.html)

**Review by chrzaszcz — APPROVED** [review]
The changes look good to me. I didn't check if unchanged files are OK, but we can do it when adding tests per category.

**Review by JanuszJakubiec — APPROVED** [review]

**Pull request merged**